modules/nlu/src/backend/pipelines/intents/ft_classifier.ts

```diff
@@ -2,20 +2,21 @@ import * as sdk from 'botpress/sdk'
 import { createWriteStream, readFileSync, writeFileSync } from 'fs'
 import _ from 'lodash'
 import tmp from 'tmp'
+import { VError } from 'verror'
 
-import { IntentClassifier } from '../../typings'
+import { IntentClassifier, IntentModel } from '../../typings'
 
 interface TrainSet {
   name: string
   utterances: Array<string>
 }
 
 export default class FastTextClassifier implements IntentClassifier {
+  private _modelsByContext: { [key: string]: sdk.MLToolkit.FastText.Model } = {}
```

Comment:
I got what you did here but i'm not sure about having multiple models in one FTClassifier...

What about keeping the FTClassifier bounded to a single FTModel + having a modelsByContext cache outside the FTclassifier which would contain FtClassifier instances. That would move the context logic out..